unit_tests/src/test_rc_data_obj.cpp

```diff
@@ -540,14 +552,206 @@ TEST_CASE("open,read,write,close")
             CHECK(contents.size() + 1 == static_cast<unsigned long>(replica_info.size()));
             CHECK(STALE_REPLICA == replica_info.replica_status());
         }
+
+        SECTION("create_empty_replica_on_non_default_resource_and_write_in_separate_open__issue_5548")
```

Comment:
@d-w-moore: These 3 unit tests are meant to kind of emulate the python scripts you provided in #5548 (original issue and follow-up issue) and #5848. Let me know if these seem equivalent to you and/or if they could use more comments to make that clearer. Thanks!

Reply:
and regardless of @d-w-moore's answer here... Dan, please make sure to write new PRC tests that box in these two issues as well.

Reply:
Wrt tests, will do .... I've now verified #5848 is fixed from the PRC perspective. Now will check 5548.

Reply:
@alanking Also wrt 5848 and clients, I've noticed that RESC_HIER_STR_KW is the operative keyword to use in conjunction with the replica token on a second open( ) call.  As opposed to  *DEST_*REST_HIER_STR_KW.  Please correct if wrong ! ...

Reply:
#5548 seems good too, statuses are updated to stale for the non-modified replica as appropriate.

Reply:
Cool, glad it's working.

Yes, `RESC_HIER_STR_KW` is the operative keyword for `rcDataObjOpen`. We could possibly allow `DESC_RESC_HIER_STR_KW` to serve the same purpose but that keyword exists for APIs such as `rcDataObjRepl` and `rcDataObjPhymv` where the user is dealing with a source and destination replica.

It does seem inconsistent that `RESC_NAME_KW` and `DEST_RESC_NAME_KW` both influence voting when resolving a resource hierarchy and yet `DEST_RESC_HIER_STR_KW` does not affect the behavior of `rcDataObjOpen` at all (that is, it does not bypass hierarchy resolution like `RESC_HIER_STR_KW` allows).

Thoughts?

Reply:
these are so poorly defined/named.  gah.

Reply:
perhaps a little markdown table here in the comments with keyword / api / comments...?

Reply:
I will be writing an issue shortly to open up this discussion elsewhere. Resolving.